openstef/feature_engineering/feature_applicator.py

```diff
@@ -46,13 +46,14 @@ def __init__(
 
     @abstractmethod
     def add_features(
-        self, df: pd.DataFrame, pj: PredictionJobDataClass
+        self, df: pd.DataFrame, pj: PredictionJobDataClass = None
     ) -> pd.DataFrame:
         """Adds features to an input DataFrame
 
         Args:
             df: pd.DataFrame with input data to which the features have to be added
-            pj (PredictionJobDataClass): Prediction job.
+            pj: (Optional) A prediction job that is needed for location dependent features,
+                if not specified a default location is used
```

Comment:
For me the expected behavior is unclear;
If pj does not have lat & lon, should a default location be used? (that's what the docstrings say).
However, the unit tests explicitly test that dni is NOT generated.

For me it would be logical that if no lat/lon is available in the pj (which is not the same as the pj being None), the additional features are not calculated.